### PR TITLE
TR-1172 - POJO Client MetadataUtils Update

### PR DESCRIPTION
**Pull request by rkreich**
- Added method getFields
- Added method getValues
- Added method getValuesByXsd
- Added method getXmlFromValues

### CODE REVIEW

**Comment by rkreich** [comment]
@tan-tan-kanarek, can you review? thanks

File: sources/pojo/src/main/java/com/kaltura/client/utils/MetadataUtils.java [hunk 1]
```diff
@@ -93,6 +100,76 @@ public static String getValue(String xml, String xPath) throws ParserConfigurati
...
+				continue;
+			
+			list.add(fieldName);
+		}
+		
+		return list;
+		
+	}
+	
+	public static LinkedHashMap<String, ArrayList<String>> getValues(String xml) throws XPathExpressionException, ParserConfigurationException, SAXException, IOException
```
Comment (tan-tan-kanarek) [inline sources/pojo/src/main/java/com/kaltura/client/utils/MetadataUtils.java]: Please change the returned type to Map<String, List<String>>, the implementation of these interfaces shouldn't matter.

File: sources/pojo/src/main/java/com/kaltura/client/utils/MetadataUtils.java [hunk 2]
```diff
@@ -93,6 +100,76 @@ public static String getValue(String xml, String xPath) throws ParserConfigurati
...
+					values.put(metadataFieldName, new ArrayList<String>());
+				
+				values.get(metadataFieldName).add(metadataFieldValue);
+			}
+		
+		}
+		return values;
+	}
+	
+	public static LinkedHashMap<String, ArrayList<String>> getValuesByXsd(String xml, String xsd) throws XPathExpressionException, ParserConfigurationException, SAXException, IOException
```
Comment (tan-tan-kanarek) [inline sources/pojo/src/main/java/com/kaltura/client/utils/MetadataUtils.java]: Please change the returned type to use interfaces.

File: sources/pojo/src/main/java/com/kaltura/client/utils/MetadataUtils.java [hunk 3]
```diff
@@ -126,15 +203,37 @@ public static void setValue(KalturaClient client, KalturaMetadata metadata, Stri
...
-		transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-		StringWriter writer = new StringWriter();
-		transformer.transform(new DOMSource(doc), new StreamResult(writer));
-		String xml = writer.getBuffer().toString().replaceAll("\n|\r", "");
+		String xml = documentToString(doc);
 
 		metadata = client.getMetadataService().update(metadata.getId(), xml);
 	}
+	
+	public static String getXmlFromValues(LinkedHashMap<String, ArrayList<String>> values) throws ParserConfigurationException, TransformerException
```
Comment (tan-tan-kanarek) [inline sources/pojo/src/main/java/com/kaltura/client/utils/MetadataUtils.java]: Change arguments type to interfaces.

**Comment by rkreich** [comment]
@tan-tan-kanarek, as discussed, `LinkedHashMap` is left because we want to force the usage of that implementation (hash map must keep order).
`ArrayList<E>` was changed to `List<E>` inteface

**Pull request merged**